### ISSUE TITLE
It is not possible to install google-cloud  0.56 due to bad dependency @ rubygems

### ISSUE THREAD

**Issue by dgpokl** [GoogleCloudPlatform/google-cloud-ruby#2332]
It's not currently possible to install the `google-cloud` gem due to a bad dependency reported by rubygems on the gems `google-cloud-container` and `google-cloud-dataproc`. Other gems in this project may also have the issue — it's hard for me to tell without inspecting every one in rubygems which I haven't completed yet.

### How to reproduce issue:

1. Create a new Gemfile containing:
```ruby
source 'https://rubygems.org'

gem 'google-cloud', '~> 0.56.0'
```
2. `gem install bundler && bundle`

#### Expected results:
Gems install.

####Actual results:
```
Bundler could not find compatible versions for gem "google-gax":
  In Gemfile:
    google-cloud (~> 0.56.0) was resolved to 0.56.0, which depends on
      google-cloud-container (~> 0.1) was resolved to 0.1.0, which depends on
        google-gax (~> 1.0.1)

    google-cloud (~> 0.56.0) was resolved to 0.56.0, which depends on
      google-cloud-text_to_speech (~> 0.1) was resolved to 0.1.0, which depends on
        google-gax (~> 1.3)
```
* Note: The first gem listed has been for me in testing sometimes google-cloud-dataproc and sometimes google-cloud-container — both have the same issue.

#### Environment tested on:
macOS 10.12.6 (16G1510)

### Analysis/Cause: 
Rubygems does not have the same dependencies listed for `google-cloud-dataproc` that the source in github declares. Compare the google-gax dependency from these two places:

* https://rubygems.org/gems/google-cloud-dataproc
* https://rubygems.org/gems/google-cloud-container

with

* https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-dataproc/google-cloud-dataproc.gemspec
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-container/google-cloud-container.gemspec

Fetching the json response from https://rubygems.org/api/v1/gems/google-cloud-dataproc.json confirms that the dependency being advertised on rubygems is on `google-gax ~> 1.0.0` or `~> 1.0.1` and not `~> 1.3` as the gemspec in github shows it is meant to be.

### Proposed solution

Since 0.1.0 has declared this bad dependency which renders it incompatible with any of the other google-cloud gems that have a dependency on a later version of `google-gax`, **I propose that version bumps be made:**

  * a bump to **0.1.1** for `google-cloud-dataproc`
  * a bump to **0.1.1** for `google-cloud-container`

**...so that the correct dependency would be picked up by rubygems and client software.**

It may be possible to update the versions 0.1.0 in-place but 1. I suspect that would be harder to do correctly and 2. anything which cached rubygems' data while this issue was present would not be guaranteed to see that a change had happened.

If this proposal is accepted, I think it would be ideal to version bump and update the gemspec of the `google-cloud` gem to specify a new minimum version for those gems, which would prevent the versions rubygems has bad information for from being used, like this:

In `google-cloud/google-cloud.gemspec`:
```ruby
  gem.add_dependency "google-cloud-container", "~> 0.1", ">= 0.1.1"
  gem.add_dependency "google-cloud-dataproc", "~> 0.1", ">= 0.1.1"
```

**Comment by dgpokl** [comment]
n.b.
`bundler` avoids this if you instruct it to install `google-cloud` without specifying a version — it will just install `0.54.0` which seems to be the last version that didn't have this impossible dependency problem. Explicitly specifying:

- just `gem 'google-cloud'` -> installs 0.54.0
- `gem 'google-cloud', '~> 0.55'` -> this error
- `gem 'google-cloud', '~> 0.56'` -> this error

**Comment by blowmage** [comment]
Thank you for the report. We have verified this and are releasing updated versions of google-cloud-container and google-cloud-dataproc.

**Comment by blowmage** [comment]
We have just released google-cloud-container 0.2.0 and google-cloud-dataproc 0.2.0. This seems to have fixed the conflicting dependencies.

```
$ gem install google-cloud
Fetching: google-cloud-bigquery-1.7.0.gem (100%)
Successfully installed google-cloud-bigquery-1.7.0
Fetching: google-cloud-bigquery-data_transfer-0.2.0.gem (100%)
Successfully installed google-cloud-bigquery-data_transfer-0.2.0
Fetching: google-cloud-bigtable-0.1.0.gem (100%)
Successfully installed google-cloud-bigtable-0.1.0
Fetching: google-cloud-container-0.2.0.gem (100%)
Successfully installed google-cloud-container-0.2.0
Fetching: google-cloud-dataproc-0.2.0.gem (100%)
Successfully installed google-cloud-dataproc-0.2.0
Fetching: google-cloud-datastore-1.4.0.gem (100%)
Successfully installed google-cloud-datastore-1.4.0
Fetching: google-cloud-dialogflow-0.1.0.gem (100%)
Successfully installed google-cloud-dialogflow-0.1.0
Fetching: google-cloud-dlp-0.5.0.gem (100%)
Successfully installed google-cloud-dlp-0.5.0
Fetching: google-cloud-dns-0.29.0.gem (100%)
Successfully installed google-cloud-dns-0.29.0
Fetching: stackdriver-core-1.3.0.gem (100%)
Successfully installed stackdriver-core-1.3.0
Fetching: google-cloud-error_reporting-0.30.1.gem (100%)
Successfully installed google-cloud-error_reporting-0.30.1
Fetching: google-cloud-firestore-0.23.0.gem (100%)
Successfully installed google-cloud-firestore-0.23.0
Fetching: google-cloud-kms-0.2.0.gem (100%)
Successfully installed google-cloud-kms-0.2.0
Fetching: google-cloud-language-0.30.0.gem (100%)
Successfully installed google-cloud-language-0.30.0
Fetching: google-cloud-logging-1.5.1.gem (100%)
Successfully installed google-cloud-logging-1.5.1
Fetching: google-cloud-monitoring-0.28.0.gem (100%)
Successfully installed google-cloud-monitoring-0.28.0
Fetching: google-cloud-os_login-0.2.0.gem (100%)
Successfully installed google-cloud-os_login-0.2.0
Fetching: google-cloud-pubsub-0.32.0.gem (100%)
Successfully installed google-cloud-pubsub-0.32.0
Fetching: google-cloud-resource_manager-0.30.0.gem (100%)
Successfully installed google-cloud-resource_manager-0.30.0
Fetching: google-cloud-spanner-1.6.0.gem (100%)
Successfully installed google-cloud-spanner-1.6.0
Fetching: google-cloud-speech-0.30.0.gem (100%)
The 0.30.0 release introduced breaking changes relative to the previous release, 0.29.0. For more details and instructions to migrate your code, please visit the migration guide: https://cloud.google.com/speech-to-text/docs/ruby-client-migration.
Successfully installed google-cloud-speech-0.30.0
Fetching: google-cloud-storage-1.13.0.gem (100%)
Successfully installed google-cloud-storage-1.13.0
Fetching: google-cloud-tasks-0.2.1.gem (100%)
Successfully installed google-cloud-tasks-0.2.1
Fetching: google-cloud-text_to_speech-0.1.0.gem (100%)
Successfully installed google-cloud-text_to_speech-0.1.0
Fetching: google-cloud-trace-0.33.1.gem (100%)
Successfully installed google-cloud-trace-0.33.1
Fetching: google-cloud-translate-1.2.0.gem (100%)
Successfully installed google-cloud-translate-1.2.0
Fetching: google-cloud-video_intelligence-1.0.0.gem (100%)
Successfully installed google-cloud-video_intelligence-1.0.0
Fetching: google-cloud-0.56.0.gem (100%)
------------------------------
Thank you for installing Google Cloud!

IMPORTANT NOTICE:
The google-cloud gem contains all the google-cloud-* gems.
Instead of depending on this gem, we encourage you to install just
the individual gems needed for your project.
------------------------------
Successfully installed google-cloud-0.56.0
28 gems installed
```

**Comment by blowmage** [comment]
Thank you @dgpokl for filing this issue!
